apps/exchange/lib/exchange/order_book.ex

```diff
@@ -277,12 +277,22 @@ defmodule Exchange.OrderBook do
 
   @spec update_queue(order_book, atom, price_in_cents, queue_of_orders) :: order_book
   def update_queue(order_book, side, price_point, new_queue) do
-    updated_side_order_book =
-      order_book
-      |> Map.fetch!(side)
-      |> Map.put(price_point, new_queue)
-
-    Map.put(order_book, side, updated_side_order_book)
+    len = Enum.count(new_queue)
+    case len do
+      0 ->
+        updated_side_order_book =
+          order_book
+          |> Map.fetch!(side)
+          |> Map.delete(price_point)
+        Map.put(order_book, side, updated_side_order_book)
+      _ ->
+        updated_side_order_book =
+          order_book
+          |> Map.fetch!(side)
+          |> Map.put(price_point, new_queue)
+
+        Map.put(order_book, side, updated_side_order_book)
+    end
```

Comment:
we  could save a few lines without losing expressiveness doing:

```elixir
   ob  =  order_book |> Map.fetch!(side)

   updated_side_order_book =
   if Enum.count(new_queue) == 0 do
      ob |> Map.delete(price_point)
    else
      ob |> Map.put(price_point, new_queue)
   end

   Map.put(order_book, side, updated_side_order_book)
```

but your refactor is already a good improvement 💯